Source/HTTP/httpcall.cpp

```diff
@@ -100,8 +101,32 @@ HRESULT CALLBACK HC_CALL::PerfomAsyncProvider(XAsyncOp op, XAsyncProviderData co
         }
         else
         {
-            // Schedule iteration 0
-            RETURN_IF_FAILED(XTaskQueueSubmitDelayedCallback(context->workQueue, XTaskQueuePort::Work, performDelay, context, HC_CALL::PerformSingleRequest));
+#if !HC_NOZLIB
+
+#if HC_PLATFORM == HC_PLATFORM_WIN32 || HC_PLATFORM == HC_PLATFORM_GDK
+            // Compress body before call if applicable
+            if (call->compressionLevel != HCCompressionLevel::None)
+            {
+                // Schedule compression task
+                RETURN_IF_FAILED(XTaskQueueSubmitDelayedCallback(context->workQueue, XTaskQueuePort::Work, performDelay, context, HC_CALL::CompressRequestBody));
+            }
+            else
+            {
+#endif
+
+#endif // !HC_NOZLIB
+
+                RETURN_IF_FAILED(XTaskQueueSubmitDelayedCallback(context->workQueue, XTaskQueuePort::Work, performDelay, context, HC_CALL::PerformSingleRequest));
+
+#if !HC_NOZLIB
+
+#if HC_PLATFORM == HC_PLATFORM_WIN32 || HC_PLATFORM == HC_PLATFORM_GDK
+
+            }
+#endif
+
+#endif // !HC_NOZLIB
```

Comment:
nit: for readability, could you break this up into a single if HC_NOZLIB preprocessor block? You'll need to duplicate one line, but I think it will be much more readable